test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Rendering/HtmlHelperDisplayExtensionsTest.cs

```diff
@@ -0,0 +1,380 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.TestCommon;
+using Microsoft.AspNetCore.Mvc.ViewEngines;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Rendering
+{
+    public class HtmlHelperDisplayExtensionsTest
+    {
+        [Fact]
+        public void DisplayHelpers_FindsModel_WhenViewDataIsNotSet()
+        {
+            // Arrange
+            var expected = $"<div class=\"HtmlEncode[[display-label]]\">HtmlEncode[[SomeProperty]]</div>{Environment.NewLine}" + 
+                $"<div class=\"HtmlEncode[[display-field]]\"></div>{Environment.NewLine}";
+            var model = new SomeModel();
```

Comment:
Set `SomeProperty` to something that's not `null`. As-is, `displayResult` would be the same if `ViewData.Model` were `null`.